### PR TITLE
Add opt-in multi-root aware file watcher based on Axosoft/nsfw

### PR DESCRIPTION
**Pull request by Tyriar**
Part of #28947 

The new WIP file watcher can be activated using the temporary `files.useNsfwFileWatcher` setting.

### CODE REVIEW

File: src/vs/workbench/services/files/node/fileService.ts [hunk 1]
```diff
@@ -120,10 +127,14 @@ export class FileService implements IFileService {
 		}
 
 		if (this.basePath && !this.options.disableWatcher) {
-			if (isWindows) {
-				this.setupWin32WorkspaceWatching();
+			if (this.options.useNsfwFileWatcher) {
```
Comment (Tyriar) [inline src/vs/workbench/services/files/node/fileService.ts]: @bpasero merging this now as the tests are passing and the new code should all be blocked off by this setting.

Comment (bpasero) [inline src/vs/workbench/services/files/node/fileService.ts]: @Tyriar sounds good. should we enable it automatically when we detect multiple folders are used?

Comment (Tyriar) [inline src/vs/workbench/services/files/node/fileService.ts]: Not just yet, I haven't tested at all on Windows. I may do so after playing with it on Monday.

Comment (Tyriar) [inline src/vs/workbench/services/files/node/fileService.ts]: @bpasero also `getWorkspace2` seemed to be returning a single root at the time of initialization. Maybe I wasn't using it right but that may complicate things.

Comment (bpasero) [inline src/vs/workbench/services/files/node/fileService.ts]: @Tyriar yes this is something I have to follow up with Sandeep, currently multi-folder comes in late.

**Pull request merged**